GameOfLife.java

```diff
@@ -84,31 +130,57 @@ private static int[][] evolve(int[][] board) {
 	// Assumes that i is at least 1 and at most the number of rows in the board - 1. 
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
 	// Uses the count(board,i,j) function to count the number of alive neighbors.
-	private static int cellValue(int[][] board, int i, int j) {
-		//// Replace the following statement with your code.
+	public static int cellValue(int[][] board, int i, int j) 
+	{
+		int sumelivecell = count(board,i,j);
+		if(board[i][j]==1)//if the cell alive
+		{
+			if(sumelivecell<1||sumelivecell>3)//check if there is over 3 live cell or less than one
+				return 0;
+				if(sumelivecell==2||sumelivecell==3)//check if there is 2 or 3 live cell next to him
+					return 1;
+		}
+		if(board[i][j]==0)//check if the cell is dead
+		{
+			if(sumelivecell==3)//if there is 3 cell live next so the cell alive
+					return 1;
+						else return 0;
+					
+		}
 		return 0;
 	}
 	
 	// Counts and returns the number of living neighbors of the given cell
 	// (The cell itself is not counted).
 	// Assumes that i is at least 1 and at most the number of rows in the board - 1. 
 	// Assumes that j is at least 1 and at most the number of columns in the board - 1. 
-	private static int count(int[][] board, int i, int j) {
+	public static int count(int[][] board, int i, int j) {
 		//// Replace the following statement with your code.
-		return 0;
+		return (board[i+1][j])+(board[i-1][j])+(board[i][j-1])+(board[i][j+1])+(board[i+1][j+1])+(board[i-1][j-1])+(board[i+1][j-1])+(board[i-1][j+1]);
```

Comment:
Better to not write such long lines of code.
You can always find a way to split it and make it more clear.